src/promptflow/promptflow/tracing/tracer.py

```diff
@@ -0,0 +1,187 @@
+import json
+import inspect
+import logging
+import uuid
+from collections.abc import Iterator
+from contextvars import ContextVar
+from datetime import datetime
+from typing import Dict, List, Optional
+
+from promptflow._core.generator_proxy import GeneratorProxy, generate_from_proxy
+from promptflow._core.thread_local_singleton import ThreadLocalSingleton
+from promptflow._utils.dataclass_serializer import serialize
+from promptflow.contracts.tool import ConnectionType
+
+from .._utils.utils import default_json_encoder
+from .contracts.trace import Trace, TraceType
+
+
+class Tracer(ThreadLocalSingleton):
```

Comment:
is this mean to public exposed? change file to _tracer if not.

Reply:
Changed to `_tracer`.